### PR TITLE
ci(ci): replace node 19 with node 20 in test matrices

### PR DESCRIPTION
**Pull request by Fdawgs**
19 becomes EOL on 2023-06-01 and 20 will be the next LTS, so makes sense.

#### Checklist

- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
CI seems to be failing :(

**Review by erfanium — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]

**Comment by erfanium** [comment]
can we drop node 14? it's not maintained anymore
https://endoflife.date/nodejs

**Comment by climba03003** [comment]
> can we drop node 14?

No, unless a new major.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -79,6 +79,10 @@ jobs:
       matrix:
         node-version: [14, 16, 18, 20]
         os: [macos-latest, ubuntu-latest, windows-latest]
+        exclude:
+          # excludes node 14 on Windows
+          - os: macos-latest
```
Comment (Uzlopak) [inline .github/workflows/ci.yml]: comment and os value are not matching

**Comment by Fdawgs** [comment]
🥳 What was the change that stopped this @mcollina ?

**Comment by mcollina** [comment]
Happy eyeballs bug in Node v20.

https://github.com/fastify/fastify/pull/4691/commits/c62dc4e3c8afea55efe7581308fca3cf94486ae7

**Comment by climba03003** [comment]
Pretty sad the `happy eyeball` issue not fixed until now. 
Already 3 release on `node@20`, hopes it get fixed before going to `LTS` (2023-10-24, 4 months left).

**Pull request merged**